Fix Auto Attacks

lzardy opened the pull request:
Fixes the issue of being able to carry over the time spent auto attacking into the next time you retarget something. Referenced in #472 
Fixes the issue of using missiles for melee auto attacks.
Fixes the issue of players being able to stop turrets from retargeting them by quickly leaving the target range after the turret begins auto attacking.
Fixes the issue of the server not properly communicating auto attack cancels to the client.
NOTE: Missile Net IDs should be fixed in the future, as when using them in LeaguePackets' packets, they do not work properly. Because of this, the missileNetID parameter of NotifyInstantStopAttack defaults to 0.

lzardy committed:
Fix auto attacks

Fixes the issue of being able to carry over the time spent auto attacking into the next time you retarget something.
Also fixes the issue of using missiles for melee auto attacks.
Also seems to fix the issue of players being able to stop turrets from retargetting them by quickly leaving the target range after the turret begins auto attacking.

lzardy committed:
Fix Submodule Pointer

danil179 commented:
Additional Information:

This PR solves a misconception in the former LS auto attack system. The PR adds tracking on the cast delay time that solves issues with repetitive auto attacks.

FurkanS1821 commented on GameServerLib/Content/CharData.cs:
```diff
@@ -110,7 +112,7 @@ public void Load(string name)
             MpRegenPerLevel = file.GetFloat("Data", "MPRegenPerLevel", MpRegenPerLevel);
             AttackSpeedPerLevel = file.GetFloat("Data", "AttackSpeedPerLevel", AttackSpeedPerLevel);
-            IsMelee = file.GetString("Data", "IsMelee", IsMelee ? "Yes" : "No").Equals("yes");
+            IsMelee = file.GetString("Data", "IsMelee", IsMelee ? "true" : "false").Equals("true");
```
Did we change something in the way we store attack data? Because in real inibins, the value is stored as "Yes" and "No".

lzardy replied:
Json files in Content store it as true or false as far as I checked.

FurkanS1821 replied:
Alright then.

FurkanS1821 reviewed (approved):
I see nothing wrong except the comment I dropped (that might be because I forgot something because of being so inactive)

Also, the commit name. That can be edited by us before it is merged too, though.

lzardy committed:
Fix AA Cancel Desyncs

Adds the packet NPC_Instant_Stop_Attack from LeaguePackets and replaces all previous uses of NotifyStopAutoAttack with NotifyInstantStopAttack.
NOTE: Missile Net IDs should be fixed in the future, as when using them in LeaguePackets' packets, they do not work properly. Because of this, the missileNetID parameter of NotifyInstantStopAttack defaults to 0.
